src/scry.cr

```diff
@@ -26,6 +29,15 @@ module Scry
       ensure
         response = Response.new([results].flatten)
         response.write(STDOUT)
+        next_request.send i
+      end
+
+      select
+      when n = next_request.receive
```

Comment:
It doesn't work, because you're blocking here!

`select` basically waits for any Channel or IO to unlock, but here you're waiting for only 1 channel.
You probably want to wait on the channels of all the requests that are currently being executed (in their own fiber) + on STDIN, so that as soon as you get a new request, STDIN will unlock, and you'll read the request, spawn a fiber etc.. And when a processing fiber finished, it writes to its channel, the channel unlocks in the select, and you can send back the result..

Not sure if it's clear enough.

Reply:
I'm a bit confused, can you show me a minimal example?

Do you mean something like this:

```crystal
loop do
  content = Request.new(STDIN).read # Is this blocking?
  spawn do
    # do something with content
  end
  # Where should the channels go?
end
```

Reply:
I think what @bew means is `select` is blocking, so it won't unblock until the channel receives a message, which is why you are seeing synchronous behavior, since it won't unblock until the request is done processing and you do `channel.send`.  You can also wait on STDIN in the select so it will unlock when a new request comes in and can spawn the new fiber there.